WordPress/gutenberg
# Improve Custom HTML Block so preview mode persists on save.

## What?
This PR is to maintain preview status on HTML Block when saved, so on refresh if the HTML block was saved in preview mode, it will load in preview mode. If it was saved in raw HTML mode, it will load in raw HTML mode.

Addresses https://github.com/WordPress/gutenberg/issues/40913

## Why?
Raw HTML is sometimes unavoidable and can be scary or jarring to the enduser. In order to maintain a what-you-see-is-what-you-get feel in the block editor, it is important to have the option to see Raw HTML in preview mode where it can look more at home with the rest of the blocks/content in the editor.

## How?
This is just a small change to the block making `preview` an attribute of the Custom HTML block. Now its state is saved with the block and can easily be loaded in preview mode or raw HTML mode.

## Testing Instructions
1. Create a new post or open an existing one.
2. Add a new or edit an existing Custom HTML Block.
3. On load it should be RAW HTML, change it to preview mode, save, and refresh.
4. It should load in preview mode. Change it to raw HTML mode and save.
5. It should load in raw HTML mode.

### Testing Instructions for Keyboard
N/A. No changes were made to the general UI of the block.
